modules/rte_mis_state/src/Routing/RouteSubscriber.php

```diff
@@ -38,6 +38,22 @@ protected function alterRoutes(RouteCollection $collection) {
       $requirements['_user_edit_access_check'] = TRUE;
       $userEditPage->setRequirements($requirements);
     }
+
+    $termAddPage = $collection->get('entity.taxonomy_term.add_form');
+    if ($termAddPage instanceof Route) {
+      // Update the title for the route.
+      $termAddDefaults = $termAddPage->getDefaults();
+      $termAddDefaults['_controller'] = 'Drupal\rte_mis_core\Controller\OverrideTaxonomyController::addForm';
+      $termAddPage->setDefaults($termAddDefaults);
+    }
+
+    $termEditPage = $collection->get('entity.taxonomy_term.edit_form');
+    if ($termEditPage instanceof Route) {
+      // Update the title for the route.
+      $termEditDefaults = $termEditPage->getDefaults();
+      $termEditDefaults['_title'] = 'Edit';
+      $termEditPage->setDefaults($termEditDefaults);
```

Comment:
@RohanSingh-inno We discussed a different approach yesterday.

Reply:
That won't work as there are two separate events in event subscriber and route subscriber, we have **RequestEvent $event** in event subscriber and **RouteBuildEvent $event**. The later ultimately returns the collections. Referrence: [RouteBuildEvent](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Routing%21RouteBuildEvent.php/class/RouteBuildEvent/10)